examples/example-gateway/endpoints/baz/baz_simpleservice_method_call_test.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// TODO: (lu) to be generated
+
+package baz
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/uber/zanzibar/test/lib/test_gateway"
+
+	"github.com/uber/zanzibar/examples/example-gateway/build/gen-code/github.com/uber/zanzibar/clients/baz/baz"
+	bazServer "github.com/uber/zanzibar/examples/example-gateway/clients/baz"
+)
+
+var testCallCounter int
+
+type testCallHandler struct {
+	*bazServer.Handler
+}
+
+func (h *testCallHandler) Call(
+	ctx context.Context, reqHeaders map[string]string, r *baz.SimpleService_Call_Args,
+) (map[string]string, *baz.BazResponse, error) {
+	testCallCounter++
+	var resp *baz.BazResponse
+	if r.Arg.B1 && r.Arg.S2 == "hello" && r.Arg.I3 == 42 {
+		resp = &baz.BazResponse{
+			Message: "yo",
+		}
+	}
+	return nil, resp, nil
+
+}
+
+func TestCallSuccessfulRequestOKResponse(t *testing.T) {
+	gateway, err := testGateway.CreateGateway(t, nil, &testGateway.Options{
+		KnownTChannelBackends: []string{"SimpleService"},
+		TestBinary: filepath.Join(
+			getDirName(), "..", "..", "build", "main.go",
+		),
+	})
+	if !assert.NoError(t, err, "got bootstrap err") {
+		return
+	}
+	defer gateway.Close()
+
+	server := bazServer.NewSimpleServiceServer(&testCallHandler{})
+	gateway.TChannelBackends()["SimpleService"].Register(server)
```

Comment:
It would be nice if we can register a single method instead of an entire service.